pkg/operator/ceph/object/objectstore.go

```diff
@@ -757,15 +759,50 @@ func enableRGWDashboard(context *Context) error {
 	if err != nil || errCode != 0 {
 		return errors.Wrapf(err, "failed to create user %q", DashboardUser)
 	}
-	args := []string{"dashboard", "set-rgw-api-access-key", *u.AccessKey}
-	cephCmd := ceph.NewCephCommand(context.Context, context.clusterInfo, args)
+
+	filePasswordSupported := mgr.FileBasedPasswordSupported(context.clusterInfo)
+	var accessArgs, secretArgs []string
+	var secretFile *os.File
+
+	// for latest Ceph versions
+	if filePasswordSupported {
+		accessFile, err := mgr.CreateTempPasswordFile(*u.AccessKey)
+		if err != nil {
+			return err
+		}
+
+		accessArgs = []string{"dashboard", "set-rgw-api-access-key", "-i", accessFile.Name()}
+		defer func() {
+			if err := os.Remove(accessFile.Name()); err != nil {
+				logger.Error(err, "failed to clean up dashboard access-key file")
+			}
+		}()
+
+		secretFile, err = mgr.CreateTempPasswordFile(*u.SecretKey)
+		if err != nil {
+			return err
+		}
+
+		secretArgs = []string{"dashboard", "set-rgw-api-secret-key", "-i", secretFile.Name()}
+		defer func() {
```

Comment:
this defer needs to be inside the goroutine around line 807, otherwise the file could be deleted before it's consumed

Reply:
honestly the defer statements may not be worth the trouble. If you just delete the files after they are used, that's fine too